gluonnlp/model/embedding.py

```diff
@@ -0,0 +1,377 @@
+        super(EmbeddingModel, self).__init__(**kwargs)
+        self.embedding_size = embedding_size
+
+    def _to_token_embedding_batch(self, batch, model_vocab, model_subwordvocab,
+                                  unknown_behavior, ctx):
+        # Handle subwords
+        subword_padding = Pad(pad_val=-1)
+        if model_subwordvocab is not None:
+            subwords = model_subwordvocab.words_to_subwordindices(batch)
+            subwords = subword_padding(subwords)
+            subwords_mask = subwords != -1
+            subwords += subwords == -1  # -1 is invalid. Change to 0
+
+            if unknown_behavior == 'impute_raise':
+                # Check that subwords are present  for all tokens
+                without_subwords_idxs = np.where(
+                    (subwords_mask.max(axis=1) == 0).asnumpy())[0].tolist()
+                if len(without_subwords_idxs):
+                    without_subwords = [
+                        batch[idx] for idx in without_subwords_idxs
+                    ]
+                    raise ValueError('No subwords were found for: ' +
+                                     ', '.join(without_subwords))
+
+            subwords = nd.array(subwords, ctx=ctx)
+            subwords_mask = nd.array(subwords_mask,
+                                     dtype=np.float32).as_in_context(ctx)
+
+        # Handle words
+        if unknown_behavior == 'raise':
+            if any(token not in model_vocab for token in batch):
+                raise ValueError
+            words = model_vocab[batch]
+            mask = nd.ones(shape=(len(words), ), ctx=ctx)
+        if unknown_behavior == 'impute_raise':
+            words = [
+                model_vocab[token] if token in model_vocab else 0
+                for token in batch
+            ]
+            mask = nd.array(
+                [1 if token in model_vocab else 0 for token in batch], ctx=ctx)
+
+        words = nd.array(words, ctx=ctx)
+
+        # Compute embeddings
+        if model_subwordvocab is not None:
+            return self(words, mask, subwords, subwords_mask)
+        else:
+            return self(words, mask)
+
+    def to_token_embedding(self, tokens, vocab, subword_vocab=None,
+                           unknown_behavior='impute_raise', batch_size=1024,
+                           ctx=cpu()):
+        """Computes a TokenEmbedding from the trained embedding model.
+
+        Parameters
+        ----------
+        tokens : list of str
+            The tokens for which to add vectors to the resulting
+            TokenEmbedding.
+        vocab : :class:`gluonnlp.Vocab` instance
+            The vocabulary of the `embedding_model`.
+        subword_vocab : :class:`gluonnlp.vocab.SubwordVocab`, optional
+            The subword vocabulary of the `EmbeddingModel`. Only needed if the
+            `EmbeddingModel` makes use of subword information.
+        unknown_behavior : ['impute_raise', 'raise'], default 'impute_raise'
```

Comment:
maybe add a fall back option too, for which the unknown token is loaded from vocab. it might be desirable to just give zeros sometimes.